### DISCUSSION TITLE
Can't renew Let's Encrypt certificate through admin panel

### DISCUSSION THREAD

**Discussion by nabiu256** [requarks/wiki#6372] [Error / Bug Report]
## Specs

- Server: DigitalOcean droplet via the Marketplace app.
- OS: Ubuntu 20.04 (LTS)
- Version: Wiki.js 2.5.298

## Problem

I created this DigitalOcean droplet some months ago so that I could migrate my WikiJS from my previous VPS provider. The setup went completely fine, and Let's Encrypt was set up properly: HTTPS was working.

Recently I received the notice that my certificate was going to expire, so I tried to renew the certificate through the SSL section in the admin panel.

I have tried multiple times, and every time I try, the same error pops up:

```
Network error: NetworkError when attempting to fetch resource.
```

After some more tries, the message changes to Let's Encrypt telling me there were too many attempts and I've been blocked from trying more in the next X hours.

## Solutions tried

I looked at the docs and tried the manual solution that they offered: ssh'ing into the droplet and restarting the `wiki` container. However, that only renovates the certificate for a few days. Is there any way to fix the admin panel issue, or to renew the certificate for longer?

EDIT: After trying again, it seems that even if the admin panel is showing me an error, the certificate _does_ get renewed, but again, just for a few days. I'm still puzzled about the error, and how to renew it for longer (or automate the process, in case it's not possible).

**Comment by jferragut** [comment]
I am experiencing the same issue. Fetch fails from Admin on Let's Encrypt renewal, and restarting container extends the date of the cert for a week. What is the solution to get a full extension of the SSL?

![image](https://github.com/requarks/wiki/assets/31458614/333162d2-b30f-469f-bf18-c255536f1126)

**Reply by whatwho** [reply]
I'm experiencing the same issue (also on Digital Ocean Droplet, running in a container). Have you found any solution for this?

**Comment by AMDphreak** [comment]
It will conduct the update but retain the old certificate until your certificate is about to expire (within 1 to 3 days). This is poorly documented behavior, with no alert that this decision is being made on behalf of the user. This needs to be fixed.
